lib/src/arithmetic/sign_magnitude_adder.dart

```diff
@@ -70,20 +87,49 @@ class SignMagnitudeAdder extends Adder {
     }
 
     final sub = aSign ^ bSign;
-    final endCarry = Logic();
-    final adder =
-        OnesComplementAdder(mux(_sign & sub, ~a, a), mux(_sign & sub, ~b, b),
-            endAroundCarry: largestMagnitudeFirst
-                ? endAroundCarry != null
-                    ? endCarry
-                    : null
-                : null,
-            subtractIn: sub,
-            adderGen: adderGen);
-    // sum <= adder.sum.getRange(0, adder.sum.width - 1).zeroExtend(sum.width);
+
+    final adder = OnesComplementAdder(
+        mux(_sign & sub, ~a, a), mux(_sign & sub, ~b, b),
+        generateEndAroundCarry: largestMagnitudeFirst & generateEndAroundCarry,
+        subtractIn: sub,
+        adderGen: adderGen);
     sum <= adder.sum;
-    if (endAroundCarry != null) {
-      output('endAroundCarry') <= (largestMagnitudeFirst ? endCarry : Const(0));
+    if (generateEndAroundCarry) {
+      endAroundCarry! <=
+          (largestMagnitudeFirst ? adder.endAroundCarry! : Const(0));
     }
   }
 }
+
+/// A sign-magnitude adder implementation that uses two ones-complement
+/// adders wired in opposition to compute the magnitude and sign without
+/// using internal twos-complement addition.
```

Comment:
is it worth adding an explanation about why you'd use this one?  it's bigger but faster?